zerver/tests/test_retention.py

```diff
@@ -177,56 +179,25 @@ def test_expired_messages_in_one_realm(self) -> None:
 
         # Only MIT has a retention policy:
         expired_msg_ids = expired_mit_msg_ids
-        move_expired_to_archive()
+        expired_usermsg_ids = self._get_usermessage_ids(expired_msg_ids)
 
-        self.assertEqual(ArchivedMessage.objects.count(), len(expired_msg_ids))
-        self.assertEqual(
-            ArchivedUserMessage.objects.count(),
-            UserMessage.objects.filter(message_id__in=expired_msg_ids).count()
-        )
-
-        self._verify_archive_data(expired_msg_ids)
+        archive_messages()
+        self._verify_archive_data(expired_msg_ids, expired_usermsg_ids)
 
         self._set_realm_message_retention_value(self.zulip_realm, ZULIP_REALM_DAYS)
 
-    """TODO: Cross realm message archiving and its testing  needs more work """
-
-    def test_cross_realm_messages_archiving_one_realm_expired(self) -> None:
-        """Test that a cross-realm message that is expired in only
-        one of the realms only has the UserMessage for that realm archived"""
-        msg_id = self._send_cross_realm_message()
-        # Make the message expired on Zulip only:
-        self._change_messages_pub_date([msg_id], timezone_now() - timedelta(ZULIP_REALM_DAYS+1))
-
-        archived_data_info = move_expired_to_archive()
-
-        self.assertEqual(ArchivedMessage.objects.count(), 1)
-        # Normally one should be archived, but we redundantly archive both usermessages,
-        # until we implement proper handling of cross-realm messages.
-        self.assertEqual(ArchivedUserMessage.objects.count(), 2)
-
-        clean_expired(archived_data_info)
-
-        self.assertEqual(UserMessage.objects.filter(message_id=msg_id).count(), 1)
-        self.assertTrue(Message.objects.filter(id=msg_id).exists())
```

Comment:
Why did you remove this test?  It seems worth having (with notes on how the behavior is currently wrong) partly as documentation of our current cross-realm message behavior.

Reply:
I just replaced these two tests with this one:
```
    def test_cross_realm_messages_not_archived(self) -> None:
        """Check that cross-realm messages don't get archived or deleted."""
```

I figured there's no more point in distinguishing the "expired in one/two realms" cases, since the same happens and it's more clear to just have a single test clearly saying "We don't touch cross-realm messages".

Reply:
OK, makes sense.